src/protocol/payload/addr.rs

```diff
@@ -9,50 +9,35 @@ use std::{
     net::{IpAddr::*, Ipv6Addr, SocketAddr},
 };
 
+use super::codec::Codec;
```

Comment:
In general, I'd consolidate the imports using absolute paths as we're already importing `protocol::payload::{...}`

Reply:
Good point, I just used RA's auto-import 💡 which did that. Fixed.